### PR TITLE
Split up tracking

### PR DESCRIPTION
**Pull request by gustaveen**
## What?

Split up different tracking services we use 

## Why?

Easier to edit and delete tracking in the future

### CODE REVIEW

**Review by styrelius — APPROVED** [review]
👌

File: src/client/utils/tracking/adTraction.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  isDanishAccidentBundle,
+  isStudentOffer,
+  isDanishTravelBundle,
+} from 'pages/OfferNew/utils'
+import { NoComboTypes, DkBundleTypes } from './tracking'
+
+type TypeOfContractExcludedUnused = Exclude<
+  TypeOfContract | NoComboTypes | DkBundleTypes,
+  'DK_ACCIDENT' | 'DK_ACCIDENT_STUDENT' | 'DK_TRAVEL' | 'DK_TRAVEL_STUDENT'
+>
```
Comment (robinandeer) [inline src/client/utils/tracking/adTraction.ts:17]: Is this a place where a comment "why" these are unused should be added?

That's a bit unclear at least to me reading this.

File: src/client/utils/tracking/adTraction.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+const adtractionProductMap: Record<TypeOfContractExcludedUnused, number> = {
```
Comment (robinandeer) [inline src/client/utils/tracking/adTraction.ts:19]: Is product the same as contract in this case? Should we agree on using only one term?

File: src/client/utils/tracking/adTraction.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    if (isDanishTravelBundle(offerData)) {
+      return isStudentOffer(offerData)
+        ? adtractionProductMap[DkBundleTypes.DkTravelBundleStudent]
+        : adtractionProductMap[DkBundleTypes.DkTravelBundle]
+    }
+  }
+  return adtractionProductMap[
+    offerData.quotes[0].contractType as TypeOfContractExcludedUnused
+  ]
```
Comment (robinandeer) [inline src/client/utils/tracking/adTraction.ts:65]: I know we do this in other place but it's also something I react to when reading. So perhaps a comment why taking the contractType from the "first item" in the list makes sense, no?

File: src/client/utils/tracking/adTraction.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+export const adtraction = (
+  orderValue: number,
+  orderId: string,
+  emailAddress: string,
+  couponCode: string | null,
+  offerData: OfferData,
+) => {
```
Comment (robinandeer) [inline src/client/utils/tracking/adTraction.ts:74]: Agree with what @styrelius usually says that so many positional arguments is less clear than accepting a single object like:

```suggestion
export const adtraction = ({
  orderValue,
  orderId,
  emailAddress,
  couponCode,
  offerData,
}: {
  orderValue: number,
  orderId: string,
  emailAddress: string,
  couponCode: string | null,
  offerData: OfferData,
}) => {
```

File: src/client/utils/tracking/tracking.ts [hunk 5]
```diff
@@ -18,6 +16,8 @@ import {
   isStudentOffer,
 } from 'pages/OfferNew/utils'
 import { trackOfferGTM } from './gtm'
+import { adtraction } from './adTraction'
```
Comment (robinandeer) [inline src/client/utils/tracking/tracking.ts:19]: Unify how we spell this word. Either camelcase or all lowercase :)

**Review by robinandeer — COMMENTED** [review]
Nice updates, just had some improvement suggestions

File: src/client/utils/tracking/adTraction.ts (hunk 1 above)
Comment (gustaveen) [reply src/client/utils/tracking/adTraction.ts:17]: Good point 🙌

File: src/client/utils/tracking/adTraction.ts (hunk 3 above)
Comment (styrelius) [reply src/client/utils/tracking/adTraction.ts:65]: We have a `getMainQuote` function in `/OfferNew/utils.ts` - maybe that could be used?

**Pull request merged**